src/main/java/org/openlmis/referencedata/domain/OrderLine.java

```diff
@@ -0,0 +1,36 @@
+package org.openlmis.referencedata.domain;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "order_lines")
+@NoArgsConstructor
+public class OrderLine {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Getter
+    @Setter
+    private Integer id;
+
+    @ManyToOne
+    @JoinColumn(name = "orderId", nullable = false)
+    @Getter
+    @Setter
+    private Order order;
+
+    @ManyToOne
+    @JoinColumn(name = "productId", nullable = false)
+    @Getter
+    @Setter
+    private Product product;
+
+    @Column(nullable = false)
+    @Getter
+    @Setter
+    private Integer orderedQuantity;
```

Comment:
For quantities, lets use Long.  This would match with v2, and would give us ample wiggle room.